Sources/core/objects/Projectiles.cpp

```diff
@@ -0,0 +1,233 @@
+//TODO maybe it can apply froce depending on bullet speed but that should be considered        ApplyForce(DirToVDir[m->dMove]);
+void Projectile::MakeMovementPattern(VDir target,id_ptr_on<Human> someone)
+{
+    set_target_=1;
+    int x = target.x;
+    int y = target.y;
+    int abs_x = std::abs(x);
+    int abs_y = std::abs(y);
+    shooter_ = someone;
+    Dir facing = someone->GetDir();
+    if (abs_x == abs_y)
+    {
+        if (y<0)
+        {
+            if (facing == 0|| facing == 1)
+            {
+                if (x>0)
+                {
+                    MakeMovementLoops(abs_y,abs_x,D_RIGHT,D_UP);
+                }
+                else
+                {
+                    MakeMovementLoops(abs_y,abs_x,D_LEFT,D_UP);
+                }
+            }
+            else
+            {
+                if (x>0)
+                {
+                    MakeMovementLoops(abs_y,abs_x,D_UP,D_RIGHT);
+                }
+                else
+                {
+                    MakeMovementLoops(abs_y,abs_x,D_UP,D_LEFT);
+                }
+            }
+        }
+        else
+        {
+            if (facing == 0|| facing == 1)
+            {
+                if ( x > 0)
+                {
+                    MakeMovementLoops(abs_y,abs_x,D_RIGHT,D_DOWN);
+                }
+                else
+                {
+                    MakeMovementLoops(abs_y,abs_x,D_LEFT,D_DOWN);
+                }
+            }
+            else
+            {
+                if (x>0)
+                {
+                    MakeMovementLoops(abs_y,abs_x,D_DOWN,D_RIGHT);
+                }
+                else
+                {
+                    MakeMovementLoops(abs_y,abs_x,D_DOWN,D_LEFT);
+                }
+            }
+        }
+    }
+    else if (x >= 0 && y >= 0)
+    {
+       if (y > x)
+       {
+            MakeMovementLoops(abs_y,abs_x,D_DOWN,D_RIGHT);
+       }
+       else if (x > y)
+       {
+            MakeMovementLoops(abs_x,abs_y,D_RIGHT,D_DOWN);
+       }
+    }
+    else if(x <= 0 && y <= 0)
+    {   
+       if ( abs_y > abs_x  )
+       {
+           MakeMovementLoops(abs_y,abs_x,D_UP,D_LEFT);
+       }
+       else if ( abs_x > abs_y)
+       {
+           MakeMovementLoops(abs_x,abs_y,D_LEFT,D_UP);
+       }
+    }
+    else if (x < 0 && y > 0)
+    {
+        if (y > abs_x)
+        {
+             MakeMovementLoops(abs_y,abs_x,D_DOWN,D_LEFT);
+        }
+        else if (abs_x > abs_y)
+        {
+            MakeMovementLoops(abs_x,abs_y,D_LEFT,D_DOWN);
+        }
+    }
+    else if (x > 0 && y < 0)
+    {
+        if (abs_y > x)
+        {
+            MakeMovementLoops(abs_y,abs_x,D_UP,D_RIGHT);
+        }
+        else if (x > abs_y )
+        {
+            MakeMovementLoops(abs_x,abs_y,D_RIGHT,D_UP);
+        }
+        
+    }
+    else
+    {
+        qDebug() << "Critical error, MakeMovementPattern has reached unreachable place";
+        abort();
+    }
+    shooter_->Rotate(movement_[1]);
```

Comment:
I do not think it is right. It is still make some dependency between projectrile and shooter. There is no much sense do so. In future there may be some changes in `MakeMovementPattern` function. 
It is really easy to implement that in `ShootImpl`
(basically, something like `GetOwner()->Rotate(VDirToDir(target))` may work - maybe some dirs will be reverted, but then it can be fixed by `DirToRDir`).